src/rules/selector-class-pattern/index.js

```diff
@@ -13,23 +14,46 @@ export const messages = ruleMessages(ruleName, {
   expected: selectorValue => `Expected class selector ".${selectorValue}" to match specified pattern`,
 })
 
-export default function (pattern) {
+export default function (pattern, options) {
   return (root, result) => {
     const validOptions = validateOptions(result, ruleName, {
       actual: pattern,
       possible: [ isRegExp, isString ],
+    }, {
+      actual: options,
+      possible: {
+        resolveNestedSelectors: isBoolean,
+      },
+      optional: true,
     })
     if (!validOptions) { return }
 
+    // In context of a nested selector, some combinations
+    // are tested mutliple times, to avoid triggering multiple
+    // identical errors, we will use an internal list of checked rules
+    var checked = []
+
     const normalizedPattern = isString(pattern) ? new RegExp(pattern) : pattern
 
     root.walkRules(rule => {
       if (cssRuleHasSelectorEndingWithColon(rule)) { return }
-      selectorParser(checkSelector).process(rule.selector)
+
+      if (options && options.resolveNestedSelectors) {
+        resolveNestedSelector(rule.selector, rule).forEach(selector => {
+          selectorParser(checkSelector).process(selector)
+        })
+      } else {
+        selectorParser(checkSelector).process(rule.selector)
+      }
 
       function checkSelector(fullSelector) {
         fullSelector.eachInside(selectorNode => {
           if (selectorNode.type !== "class") { return }
+
+          var compareableSelector = omitBy(omit(selectorNode, ["parent"]), isFunction)
```

Comment:
Seems like a good place for a comment in the code :D 
Actually, I create a clone of the `selectorNode` in a way that is comparable (without methods nor "parents")
So if two selectors are identical (not just duplicate, but exactly the same selector in the source) it will treat them as equals, and not check them twice